lib/package/kinesis/compileMethodsToKinesis.js

```diff
@@ -147,27 +138,16 @@ module.exports = {
   },
 
   buildDefaultKinesisRequestTemplate(http) {
-    let streamName
-    if (typeof http.streamName == 'object') {
-      streamName = http.streamName
-    } else {
-      streamName = `"${http.streamName}"`
-    }
-
     const objectRequestParam = this.getKinesisObjectRequestParameter(http)
 
     return {
-      'Fn::Join': [
-        '',
-        [
-          '{',
-          '"StreamName": "',
-          streamName,
-          '",',
-          '"Data": "$util.base64Encode($input.body)",',
-          `"PartitionKey": "${objectRequestParam}"`,
-          '}'
-        ]
+      'Fn::Sub': [
+        '{"StreamName":"${StreamName}","Data":"${Data}","PartitionKey":"${PartitionKey}"}',
+        {
+          StreamName: http.streamName,
```

Comment:
the old code had an if-else for the stream name and when it's a string it put `"` around the literal and the original `Fn::Join` would have put another `"` around it (e.g. `{ "StreamName": ""myStream"", "Data": ... }`. 
I'm not sure if that's correct behaviour, but the new code looks like it's gonna behave differently. Am I understanding this correctly?

Reply:
Yup, all values eventually had to convert into string in the old code. but it has became unnecessary in the new one.